kafka/source/pkg/reconciler/kafkasource.go

```diff
@@ -103,6 +109,12 @@ type Reconciler struct {
 	metricsConfig  *metrics.ExporterOptions
 
 	sinkResolver *resolver.URIResolver
+
+	minReplicaCount     *int32
```

Comment:
seems weird to have this source specific properties being set in the Reconciler object. 
I think you can remove these properties from here, and just read the values from the annotations when you construct the ScaledObject, which should have been validated in the webhook.

Reply:
do you have a digram showing how webhooks, controllers, and KEDA work together in this setup?

Reply:
no diagrams, sorry, just the code that I pointed to in the other comments

Reply:
an example of validation that I'm doing in the webhook, as well as setting up defaults... 
https://github.com/google/knative-gcp/blob/master/pkg/apis/duck/v1alpha1/annotations.go

Those functions are being called from `<source>_defaults.go` or `<source>_validation.go`... 
I see that you have a kafka_validation, which seems to be doing an immutable validation only (which also sounds weird)... and no defaults.go... Do you guys have a webhook for kafka?

Reply:
and I want to stress the fact that those variables shouldn't be in the Reconciler object. Concurrent accesses to this controller because for example two sources are being created, will mess things up...